src/Core/Config/Util/ConfigFileTransformer.php

```diff
@@ -26,67 +26,191 @@
  */
 class ConfigFileTransformer
 {
+	/**
+	 * The public method to start the encoding
+	 *
+	 * @param array $data A full config array
+	 *
+	 * @return string The config stream, which can be saved
+	 */
 	public static function encode(array $data): string
 	{
+		// Add the typical header values
 		$dataString = '<?php' . PHP_EOL . PHP_EOL;
-		$dataString .= 'return [' . PHP_EOL;
+		$dataString .= 'return ';
 
-		$categories = array_keys($data);
+		$dataString .= static::extractArray($data);
 
-		foreach ($categories as $category) {
-			if (is_null($data[$category])) {
-				$dataString .= "\t'$category' => null," . PHP_EOL;
-				continue;
-			}
-
-			$dataString .= "\t'$category' => [" . PHP_EOL;
+		// the last array line, close it with a semicolon
+		$dataString .= ";" . PHP_EOL;
 
-			if (is_array($data[$category])) {
-				$keys = array_keys($data[$category]);
+		return $dataString;
+	}
 
-				foreach ($keys as $key) {
-					$dataString .= static::mapConfigValue($key, $data[$category][$key]);
-				}
-			}
-			$dataString .= "\t]," . PHP_EOL;
+	/**
+	 * Extracts an inner config array.
+	 * Either as a Key => Value pair array or as an assoziative array
```

Comment:
In English, associative == key => value.
```suggestion
	 * Either as a list or as an associative array
```